elements/lisk-codec/src/json_wrapper.ts

```diff
@@ -56,7 +56,7 @@ const mappers: mappersInterface = {
 /* eslint-enable @typescript-eslint/explicit-function-return-type */
 
 const findObjectByPath = (message: SchemaProps, pathArr: string[]): SchemaProps | undefined => {
-	let result: SchemaProps = message;
+	let result: SchemaProps = objects.cloneDeep(message);
```

Comment:
if we need to clone the object recursively, wouldn't it be very slow?

Reply:
I am not sure about exact/future usage of this function, but current implementation is kind of changing the value of the function parameter, which is an anti-pattern. 

https://github.com/LiskHQ/lisk-sdk/blob/b0811447013466171f2d63ab813125cb8e41f658/elements/lisk-codec/src/json_wrapper.ts#L68

BTW this function `findObjectByPath` is only used in the json-wrapper so would not actually have any impact on actual encoding/decoding part.